### PR TITLE
Add more visibility to decor.

### LINKED ISSUE #1577: Arbor Python objects for decor and cell global properties present opaque interfaces
`arbor.decor` or `arbor.cell_global_property` objects have no methods or attributes that can be used to retrieve any information about them. In principle, a `decor` object can be queried for its lists of decorations, and similarly `cell_global_property` should be able to make available its defaults for ions and electrical properties.

Example use case:
```
p = arbor.neuron_cable_properties()
# How do I discover what the NEURON defaults are at this point?
```

### LINKED ISSUE #1578: Cannot remove default ions from Arbor python `cable_global_properties` object
There is no interface in Python to remove ions from the `ion_species` map in the corresponding `arb::cable_cell_global_properties`  object. This is a bit of a pain if you want to create a global properties object from scratch in a model which doesn't use one or more of the default ion species.

The process for adding new ions could be separated from that for setting their default concentrations etc. The ion definition table, mapping ion names to their charge, could be presented as a dictionary via pybind11 for example.

### LINKED ISSUE #1607: Catching bad cell behaviour in Python
Quoting from dev chat:

> In the C++ API, there is a field called `membrane_voltage_limit_mV` in the `cable_cell_global_properties` struct that can be used to throw an exception when the absolute value of the voltage exceeds the given limit (or is NaN).

But this field isn't exposed in the Python API, so there's no way to request that the simulation terminate with some sort of error when everything goes wonky.

### PR DESCRIPTION
**Pull request by thorstenhater**
Functionality to
- delete ions from global properties
- peek into global properties
- expose painted, placed, and defaulted items
- expose voltage limit to python
- make voltage limit an optional to allow for arbitrary values.
- Add ion_settings view to python = (valence, ion_data, rev_pot_method)

Documentation
- cable cell properties in python

Closes #1578 
Closes #1577 
Closes #1607

### CODE REVIEW

**Review by AdhocMan — CHANGES_REQUESTED** [review]
Just a few comments that need updating as well

File: arbor/fvm_lowered_cell_impl.hpp [hunk 1]
```diff
@@ -96,7 +96,7 @@ class fvm_lowered_cell_impl: public fvm_lowered_cell {
     std::vector<mechanism_ptr> revpot_mechanisms_;
 
     // Non-physical voltage check threshold, 0 => no check.
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply arbor/fvm_lowered_cell_impl.hpp:98]: Good catch.

**Review by AdhocMan — APPROVED** [review]
Looks good to me.
I'll leave the merge commit message to you on this one.

**Review by brenthuisman — CHANGES_REQUESTED** [review]
Missing updates to Python API, e.g. `python/decor.rst`.

**Review by brenthuisman — APPROVED** [review]

**Pull request merged**